lib/cc/cli.rb

```diff
@@ -17,15 +17,25 @@ module CLI
     autoload :ValidateConfig, "cc/cli/validate_config"
     autoload :Version, "cc/cli/version"
 
-    def self.debug(message, values = {})
-      if ENV["CODECLIMATE_DEBUG"]
-        if values.any?
-          message << " "
-          message << values.map { |k, v| "#{k}=#{v.inspect}" }.join(" ")
+    class Logger
+      def debug(message, values = {})
+        if ENV["CODECLIMATE_DEBUG"]
+          if values.any?
+            message << " "
+            message << values.map { |k, v| "#{k}=#{v.inspect}" }.join(" ")
+          end
+
+          # Leading and trailing newlines to prevent mangling
+          $stderr.print("\n[DEBUG] #{message}\n")
         end
+      end
 
-        $stderr.puts("[DEBUG] #{message}")
+      # Any non-DEBUG output is handled not via logging
+      def method_missing(*)
```

Comment:
There are other calls to `Analyzer.logger.{warn,info}` reachable by the CLI, so this shim is necessary. The interface mismatch is worth thinking about though.